src/Tests.cpp

```diff
@@ -29,60 +29,89 @@ TriGram text4_pack(const char (&s)[5]) {
     return (v0 << 18U) | (v1 << 12U) | (v2 << 6U) | (v3 << 0U);
 }
 
-Query do_select(std::string query_text) {
+template <typename T>
+T parse(std::string query_text) {
     Command cmd = parse_command(query_text);
-    REQUIRE(std::holds_alternative<SelectCommand>(cmd));
-    return std::get<SelectCommand>(cmd).get_query();
+    REQUIRE(std::holds_alternative<T>(cmd));
+    return std::get<T>(cmd);
+}
+
+Query do_select(std::string query_text) {
+    return parse<SelectCommand>(query_text).get_query();
 }
 
 QString mqs(const std::string &str) {
     QString out;
-
     for (const auto &c : str) {
-        if (c != '?') {
-            out.emplace_back(QTokenType::CHAR, c);
-        } else {
-            out.emplace_back(QTokenType::WILDCARD);
-        }
+        out.emplace_back(QTokenType::CHAR, c);
     }
 
     return out;
 }
 
-TEST_CASE("pack", "[pack]") {
+TEST_CASE("packing 3grams", "[internal]") {
     // pay attention to the input, this covers unexpected sign extension
     REQUIRE(gram3_pack("\xCC\xBB\xAA") == (TriGram)0xCCBBAAU);
     REQUIRE(gram3_pack("\xAA\xBB\xCC") == (TriGram)0xAABBCCU);
     REQUIRE(gram3_pack("abc") == (TriGram)0x616263);
 }
 
-TEST_CASE("select simple", "[queryparser]") {
-    Query q = do_select("select \"test\";");
-    REQUIRE(q.get_type() == QueryType::PRIMITIVE);
-    REQUIRE(q.as_string_repr() == "test");
+TEST_CASE("select hexstring without spaces", "[queryparser]") {
+    Query query = do_select("select{4d534d};");
+    REQUIRE(query == q(mqs("MSM")));
 }
 
-TEST_CASE("select hex lowercase", "[queryparser]") {
-    Query query = do_select("select {4d534d};");
+TEST_CASE("select hexstring with spaces", "[queryparser]") {
+    Query query = do_select("select { 4d 53 4d };");
     REQUIRE(query == q(mqs("MSM")));
 }
 
-TEST_CASE("select hex uppercase", "[queryparser]") {
-    Query query = do_select("select {4D534D};");
+TEST_CASE("select hexstring with mixed spaces", "[queryparser]") {
+    Query query = do_select("select { 4d 534d };");
     REQUIRE(query == q(mqs("MSM")));
 }
 
-TEST_CASE("select hex spaces", "[queryparser]") {
-    Query query = do_select("select {4d 53 4d};");
+TEST_CASE("select hexstring with full wildcard", "[queryparser]") {
+    QString expect;
+    expect.emplace_back(QTokenType::CHAR, 0x4d);
+    expect.emplace_back(QTokenType::WILDCARD);
+    expect.emplace_back(QTokenType::CHAR, 0x4d);
```

Comment:
Though, I'd prefer to do it with initializer list like:
```c++
    QString expect = {
        {QTokenType::CHAR, 0x4d},
        {QTokenType::WILDCARD},
        {QTokenType::CHAR, 0x4d},
    };
```
;)

Reply:
:thumbsup: though maybe not for this PR :thinking:. I'll create more thests in the future.